### PR TITLE
[LFRic] Move names of infrastructure modules to LFRicConstants (towards #1258)

### PR DESCRIPTION
**Pull request by TeranIvy**
This PR adds mappings of LFRic infrastructure types/data to their LFRic infrastructure modules to LFRic constants, so they are not hard-coded any more.

I followed the existing example of `LFRicConstants.QUADRATURE_TYPE_MAP` that uses internal PSyclone names for quadrature types and adds information about module, type and proxy type names (the latter two where available).

### CODE REVIEW

File: src/psyclone/domain/lfric/lfric_constants.py [hunk 1]
```diff
@@ -62,13 +63,13 @@ def __init__(self):
         LFRicConstants.HAS_BEEN_INITIALISED = True
         api_config = Config.get().api_conf("dynamo0.3")
```
Comment (arporter) [inline src/psyclone/domain/lfric/lfric_constants.py:64]: Please could you disable the pylint 'too-many-statements' warning for this constructor.

**Review by arporter — CHANGES_REQUESTED** [review]
A nice PR - thanks for keeping it short.
All tests and examples are OK with compilation.
We've gained one new pylint warning because of the additional statements but that can be disabled.
Documentation does not require updating.
I did wonder whether it would be better to introduce a namedtuple with `module`, `type` and `proxy_type` properties but decided it wouldn't buy us much. We could always do this in the future if we think it would be better.
There's just one other minor thing to fix.

File: src/psyclone/domain/lfric/lfric_constants.py [hunk 2]
```diff
@@ -283,6 +269,86 @@ def __init__(self):
...
+        # ---------- Infrastructure module maps -------------------------------
+
+        # Dictionary allowing us to look-up the name of the Fortran module,
+        # type and proxy-type associated with each LFRic data type.
+        LFRicConstants.DATA_TYPE_MAP = {
+            # 'real'-valued scalar of kind 'r_def' (used for global
+            # reductions of "field_type" data)
+            "scalar": {"module": "scalar_mod",
+                       "type": "scalar_type",
+                       "proxy_type": "field_proxy_type"},
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/domain/lfric/lfric_constants.py:281]: Oops, it doesn't - copy and paset gone wrong! Removed.

File: src/psyclone/domain/lfric/lfric_constants.py (hunk 1 above)
Comment (TeranIvy) [reply src/psyclone/domain/lfric/lfric_constants.py:64]: Disabled.

**Comment by TeranIvy** [comment]
Addressed review comments, back to @arporter.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Tests and examples are OK, even after merging latest master.
Will proceed to merge.

**Pull request merged**